Userland/Games/GameOfLife/BoardWidget.cpp

```diff
@@ -146,9 +148,10 @@ void BoardWidget::paint_event(GUI::PaintEvent& event)
 
             if (m_selected_pattern != nullptr) {
                 int y_offset = 0;
-                for (auto line : m_selected_pattern->pattern()) {
+                for (unsigned y = 0; y < m_selected_pattern->height(); y++) {
```

Comment:
Range-based for is much nicer, why change this?

Reply:
The first reviewer suggested not to use a Vector of strings (each redundantly storing a string-length) to store the pattern for a game of life figure but to use a data structure similar to CharacterBitmap holding only one string for the whole pattern instead.

This also means iterating over the pattern using "for (auto line : pattern)" does not work anymore. Since the Pattern class does not provide any way for accessing "the i-th line of the pattern". You can only access single characters by calling at(x, y) for which you need to know both arguments x and y.